### PR TITLE
docs(format): include API docs for each language/tool

### PR DESCRIPTION
**Pull request by alexeagle**
Also remove any default tools for languages, forcing users to explicitly list an attribute and a tool value for each.

---

### Test plan

CI

### CODE REVIEW

File: example/tools/format/BUILD.bazel [hunk 1]
```diff
@@ -54,21 +54,26 @@ format_multirun(
     name = "format",
     cc = "@llvm_toolchain_llvm//:bin/clang-format",
     css = ":prettier",
+    go = "@aspect_rules_lint//format:gofumpt",
```
Comment (gregmagolan) [inline example/tools/format/BUILD.bazel:57]: could use `BUILTIN_TOOL_LABELS["go"]` instead?

Comment (gregmagolan) [reply example/tools/format/BUILD.bazel:57]: oh.. nvmind.. I see how this is support to work now :)

**Review by gregmagolan — APPROVED** [review]
🦖

File: format/defs.bzl [hunk 2]
```diff
@@ -55,30 +63,50 @@ def _format_attr_factory(target_name, lang, toolname, tool_label, mode):
...
-    Also produces a target `[name].check` which does not edit files, rather it exits non-zero
-    if any sources require formatting.
+    This macro produces a target named `[name].check` which does not edit files,
+    rather it exits non-zero if any sources require formatting.
 
-    Tools are provided by default for some languages.
-    These come from the `@multitool` repo.
-    Under --enable_bzlmod, rules_lint creates this automatically.
-    WORKSPACE users will have to set this up manually. See the release install snippet for an example.
+    Not recommended: to check formatting with `bazel test`, use [format_test](#format_test) instead.
```
Comment (mattnworb) [inline format/defs.bzl:100]: does this mean to say that `[name].check` is not recommended, or that `format_test` is not recommended? Its pretty ambiguous

Comment (alexeagle) [reply format/defs.bzl:100]: thanks, I'll clarify it's the latter

**Pull request merged**